### PR TITLE
Upgrade to Sentry 8

### PR DESCRIPTION
**Pull request by nwalters512**
Sentry v8 made some [rather invasive changes](https://docs.sentry.io/platforms/javascript/guides/node/migration/v7-to-v8/). Specifically, they removed the very convenient `Sentry.Handlers.requestHandler()` that could be dropped easily into an Express application. Now, instrumentation is "automagical" and requires one to jump through some hoops to get it to work, [particularly if you're running native ESM](https://docs.sentry.io/platforms/javascript/guides/node/install/esm/). In our case, it would require a pretty invasive restructuring of the bootstrapping process to ensure that Sentry could put its hooks into everything when it wants to.

Rather than jump through those hoops, I pieced together a replacement `requestHandler()` middleware that gets us more or less the same experience we had with Sentry v7.

### CODE REVIEW

File: apps/prairielearn/src/lib/sentry.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import * as Sentry from '@prairielearn/sentry';
 export function enrichSentryEventMiddleware(req: Request, res: Response, next: NextFunction) {
   // This will ensure that this middleware is always run in an isolated
   // context so that we don't accidentally leak data between requests.
-  Sentry.runWithAsyncContext(() => {
+  Sentry.withScope(() => {
     const scope = Sentry.getCurrentScope();
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/lib/sentry.ts:9]: I believe, from the docs, that you can do this:
```suggestion
  Sentry.withScope((scope) => {
```

File: apps/prairielearn/package.json [hunk 2]
```diff
@@ -51,8 +51,6 @@
     "@prairielearn/signed-token": "workspace:^",
     "@prairielearn/tsconfig": "workspace:^",
     "@prairielearn/workspace-utils": "workspace:^",
-    "@sentry/profiling-node": "^7.118.0",
-    "@sentry/tracing": "^7.118.0",
```
Comment (nwalters512) [inline apps/prairielearn/package.json:55]: We haven't had these enabled in ages; I opted to just rip them out.

File: apps/prairielearn/src/lib/sentry.ts (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/src/lib/sentry.ts]: Done in 2d746a2.

**Review by mwest1066 — APPROVED** [review]
Looks like a great strategy to me

**Pull request merged**